lib/pusher-platform/sdk_info.rb

```diff
@@ -0,0 +1,26 @@
+require_relative './error'
+
+module PusherPlatform
+  class SDKInfo
+    attr_reader :product_name, :version, :language, :platform, :headers
+
+    def initialize(options)
+      raise Error.new('No product_name provided to SDKInfo') if options[:product_name].nil?
+      raise Error.new('No version provided to SDKInfo') if options[:version].nil?
+
+      @product_name = options[:product_name]
```

Comment:
Do we use "product" in other SDKs? I was wondering about the distinction between the product and the service.

Reply:
Product is indeed used in other SDKs, e.g. https://github.com/pusher/pusher-platform-js/blob/master/src/base-client.ts#L34-L36

Service is used in the `BaseClient` initialisers in the form of `serviceName` and `serviceVersion`, e.g. `chatkit-files` and `v1`, so having `service` again felt like overloading terminology. Plus `/services` is the first part of any path to a service running on the platform

Reply:
Makes sense, let's stick with the established terminology.